docs: Document Tag Manager with enhanced search capability (#701)
mautic/user-documentation: docs/plugins/tag_manager.rst

Tag Manager
###########

.. vale on

Tags help you organize and categorize your Contacts. Unlike Segments, Tags are labels you manually assign to Contacts. You can add or remove them anytime. Use Tags to mark Contacts based on characteristics, behaviors, or other criteria that matter to your marketing.

> The Tag Manager is primarily for reviewing existing Tags in your Mautic instance and deleting Tags you no longer need. You can create Tags directly in the Tag input field anywhere it appears. This includes editing a Contact, configuring a Campaign action, or setting up a Form action. Type a Tag name and press Enter. If the Tag already exists, you can select it. If it doesn't exist, Mautic creates it automatically.

Reviewer: @promptless-for-oss, we can tighten the 2nd sentence since the next one lists the specific locations.

```suggestion
The Tag Manager is primarily for reviewing existing Tags in your Mautic instance and deleting Tags you no longer need. You can create Tags directly within any Tag input field. This includes editing a Contact, configuring a Campaign action, or setting up a Form action. Type a Tag name and press Enter. If the Tag already exists, you can select it. If it doesn't exist, Mautic creates it automatically.
```

Tag Manager
###########

.. vale on

Tags help you organize and categorize your Contacts. Unlike Segments, Tags are labels you manually assign to Contacts. You can add or remove them anytime. Use Tags to mark Contacts based on characteristics, behaviors, or other criteria that matter to your marketing.

.. vale off

Accessing the Tag Manager
*************************

.. vale on

> Go to **Contacts > Manage Tags** to open the Tag Manager list page. This page shows all existing Tags with their names, descriptions, and Contact counts.

Reviewer: ```suggestion
.. vale off

Go to **Contacts > Manage Tags** to open the Tag Manager list page. This page shows all existing Tags with their names, descriptions, and Contact counts.

.. vale on
```

> Go to **Contacts** > **Manage Tags** to open the Tag Manager list page. This page shows all existing Tags with their names, descriptions, and Contact counts.

Reviewer: Let's bold all navigation instruction for consistency.

```suggestion
Go to **Contacts > Manage Tags** to open the Tag Manager list page. This page shows all existing Tags with their names, descriptions, and Contact counts.
```

.. vale off

Creating a Tag
**************

.. vale on

#. Click the **New** button in the top right corner.
#. Enter a **Name** for the Tag. This is the only required field.
#. Optionally, add a **Description** to provide more context about when to use this Tag.
#. Click **Save & Close** to create the Tag.

.. vale off

Editing a Tag
*************

.. vale on

#. Click on the Tag name in the list to open it.
#. Modify the name or description as needed.
#. Click **Save & Close** to apply changes.

.. vale off

> 

Author: PR #16003 adds the ability to search Tags by description text in addition to name. The PR enhances both the API collection search and the Tag Manager list search in the UI to match descriptions.

Source: https://github.com/mautic/mautic/pull/16003

Deleting a Tag
**************

.. vale on

#. Select the checkbox next to the Tag you want to delete.
#. Click the dropdown arrow next to the **New** button and select **Delete Selected**.
#. Confirm the deletion.

.. note::

   Deleting a Tag doesn't delete the Contacts associated with it. The Tag is simply removed from those Contacts.

.. vale off

Searching Tags
**************

.. vale on

Use the search box at the top of the Tag Manager list to find specific Tags. Mautic searches both the Tag name and description. This helps you find Tags even if you only remember part of the description.

> For example, you have a Tag named 'VIP' with the description "High-value customers who purchased premium plans." You can find it by searching for 'VIP', 'premium', or 'high-value'.

Reviewer: Let's add "VIP" to accept.txt to address Vale warning here.

> For example, say you have a Tag named 'VIP' with the description 'High-value customers who purchased premium plans'. You can find it by searching for 'VIP', 'premium', or 'high-value'.

Reviewer: Let's adjust the wording here for clarity and use double quotes for quote sentence.

```suggestion
For example, you have a Tag named 'VIP' with the description "High-value customers who purchased premium plans." You can find it by searching for 'VIP', 'premium', or 'high-value'.
```

.. vale off

Assigning Tags to Contacts
**************************

.. vale on

You can assign Tags to Contacts in several ways:

> * **Contact detail page** - Open a Contact, go to the 'Tags' tab, and add or remove Tags directly.

Reviewer: Use bold instead of single quote for UI display for consistency.

```suggestion
* **Contact detail page** - Open a Contact, go to the **Tags** tab, and add or remove Tags directly.
```

* **Batch actions** - Select multiple Contacts in the Contact list, then use the bulk action menu to add or remove Tags.

> * **Campaign actions** - Use the 'Modify Contact Tags' action in Campaigns to automatically add or remove Tags based on Contact behavior.

Reviewer: ```suggestion
* **Campaign actions** - Use the **Modify Contact Tags** action in Campaigns to automatically add or remove Tags based on Contact behavior.
```

> * **Contact detail page** - Open a Contact, go to the **Tags** tab, and add or remove Tags directly.
> * **Batch actions** - Select multiple Contacts in the Contact list, then use the bulk action menu to add or remove Tags.
> * **Campaign actions** - Use the **Modify Contact Tags** action in Campaigns to automatically add or remove Tags based on Contact behavior.
> * **Form actions** - Configure a Form to add Tags to Contacts when they submit it.
> * **Tracking pixel** - Pass Tags via the tracking URL using the ``tags`` parameter.

Reviewer: Let's wrap this in vale on/off statements.

```suggestion
.. vale off

* **Contact detail page** - Open a Contact, go to the **Tags** tab, and add or remove Tags directly.
* **Batch actions** - Select multiple Contacts in the Contact list, then use the bulk action menu to add or remove Tags.
* **Campaign actions** - Use the **Modify Contact Tags** action in Campaigns to automatically add or remove Tags based on Contact behavior.
* **Form actions** - Configure a Form to add Tags to Contacts when they submit it.
* **Tracking pixel** - Pass Tags via the tracking URL using the ``tags`` parameter.

.. vale on
```

> For more information on using Tags with the tracking pixel, see :ref:`contacts/manage_contacts:Tags`.

Reviewer: The link doesn't work. We need to add custom target before using `:ref:`. This is what you need to do:

1. In the `docs/contacts/manage_contacts.rst` file, add `.. _tracking pixel query Tags:` right above the title of "Tags" section in "Tracking pixel query".

  ```rst
  .. _tracking pixel query Tags:

  Tags
  ~~~~

  You can change the Contact's Tags by using the ``tags`` query parameter. You can separate multiple Tags by comma. To remove a Tag, prefix it with a dash (minus sign).

  For example, ``mtracking.gif?tags=ProductA``,-ProductB would add the ProductA Tag to the Contact and remove ProductB.
  ```

2. Update the `:ref:`  to point to this custom target.

  ```suggestion
  For more information on using Tags with the tracking pixel, see :ref:`Tracking pixel query <tracking pixel query Tags>` section.
  ```

Tag Manager
###########

.. vale on

Tags help you organize and categorize your Contacts. Unlike Segments, Tags are labels you manually assign to Contacts. You can add or remove them anytime. Use Tags to mark Contacts based on characteristics, behaviors, or other criteria that matter to your marketing.

.. vale off

Accessing the Tag Manager
*************************

.. vale on

.. vale off

Go to **Contacts > Manage Tags** to open the Tag Manager list page. This page shows all existing Tags with their names, descriptions, and Contact counts.

.. vale on

.. vale off

Creating a Tag
**************

.. vale on

#. Click the **New** button in the top right corner.
#. Enter a **Name** for the Tag. This is the only required field.
#. Optionally, add a **Description** to provide more context about when to use this Tag.

> #. Click **Save & Close** to create the Tag.

Reviewer 2: That's not how many users are creating tags. You can go directly to any contact edit page, find the tag input and start typing a tag name. If it doesn't exist you can hit Enter and the tag is created. Now if you start typing the same name in another contact edit page in the tag input it will find the existing tag and users can select it instead of creating a new one.

This works not only in the contact edit page but also anywhere else where the tag input is. For example in a campaign action, form action and so on.

This is very simple and users can create a tag when they need it without going to the tag manager first. The tag manager is usually used for reviewing what tags exist and deleting those that aren't useful anymore. 

I don't think many people are creating tags in the tag manager as it's way simpler to create them in the tag input.

Tag Manager
###########

.. vale on

Tags help you organize and categorize your Contacts. Unlike Segments, Tags are labels you manually assign to Contacts. You can add or remove them anytime. Use Tags to mark Contacts based on characteristics, behaviors, or other criteria that matter to your marketing.

> The Tag Manager is primarily for reviewing existing Tags in your Mautic instance and deleting Tags you no longer need. Most users create Tags inline - directly in the Tag input field when editing a Contact, configuring a Campaign action, setting up a Form action, or anywhere else the Tag input appears. Type a Tag name and press Enter. If the Tag exists, you can select it; if not, Mautic creates it automatically.

Reviewer: We need to address the Vale warnings here and reword this a bit. I feel like mentioning "most user" here is unnecessary. We want to tell all users that they can directly do so withing tag field.

```suggestion
The Tag Manager is primarily for reviewing existing Tags in your Mautic instance and deleting Tags you no longer need. You can create Tags directly in the Tag input field anywhere it appears. This includes editing a Contact, configuring a Campaign action, or setting up a Form action. Type a Tag name and press Enter. If the Tag already exists, you can select it. If it doesn't exist, Mautic creates it automatically. If the Tag exists, you can select it. If it doesn't exist, Mautic creates it automatically.
```

> For details on inline Tag creation and other ways to work with Tags, see :ref:`Tags <contacts/tags:Tags>`.

Reviewer: Fix the link that's not rendered and address Vale warning.

```suggestion
.. vale off

For details on Tag creation and other ways to work with Tags, see :doc:`/contacts/tags`.

.. vale on
```
